### PR TITLE
tools: use golangci-lint in the officially recommended way

### PR DESCRIPTION
**Pull request by Poytr1**
UCP #2181
Signed-off-by: Poytr1 <pooytr1@gmail.com>

### What problem does this PR solve? 

Resolve: https://github.com/pingcap/pd/issues/2181
Currently, golangci-lint is installed in retool, which is not an officially recommended installation method. We need to change to the officially recommended way and add the .golangci.yml configuration file.

### What is changed and how it works?

- Delete `golangci-lint` in the `retool-install.sh` as well as `tool.json`
- Add a new task in Makefile called `golangci-lint-setup` to download `golangci-lint` binary into *$GOPATH/bin/golangci-lint* as https://github.com/golangci/golangci-lint#install suggests.
- Add a `.golangci.yml` file which has same config with the command line option.
- Update the command to call `golangci-lint` using the new config file.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Run `make check`, then the `golangci-lint` will be downloaded and ran with the config file `.golangci.yml`.

Code changes

 - Has configuration change

Side effects
N/A

Related changes

 - Need to update the documentation

### CODE REVIEW

**Comment by Poytr1** [comment]
Looks like in the ci environment, the `gopath` wasn't set, should I use a path inside the project like `.tools/` to store the binary?

**Comment by nolouch** [comment]
@Poytr1 I think it's ok to use a path like `.tools/bin`

File: Makefile [hunk 1]
```diff
@@ -106,7 +106,10 @@ retool-setup:
 	@which retool >/dev/null 2>&1 || go get github.com/twitchtv/retool
 	@./scripts/retool sync
 
-check: retool-setup check-all check-plugin
+golangci-lint-setup:
+	@which golangci-lint >/dev/null 2>&1 || curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.23.7
```
Comment (HunDunDM) [inline Makefile]: I think it can be placed in the `./bin` directory. This directory can also be used as `GOBIN` for tools.go

Comment (Poytr1) [reply Makefile]: Done. This pr has conflicts with #2196, it would be better to merge #2196 firstly then I will resolve  these conflicts.

Comment (loxp) [reply Makefile]: I think it's better to merge this PR first. If merge #2196 first, it must manage golangci-lint with tools.go, which import several dependencies, and have to remove them after your PR is merged.

**Pull request merged**